### PR TITLE
Add details for TCP log collection

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

The behavior of the TCP log collection support has changed: it no longer
disconnects idle connections after 60s.  This behavior was not
documented.

This change adds documentation for the new behavior, even though it
matches the behavior a user might assume.  It also suggests
re-connecting on connection failure, as a reminder that TCP connections
in general are not failure-proof.

### Preview
https://docs-staging.datadoghq.com/dustin.mitchell/ac-76/agent/logs/?tab=tailfiles

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/agent/logs/_index.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ logs:
 
 {{% tab "TCP/UDP" %}}
 
-To gather logs from your `<APP_NAME>` application that forwards its logs with TCP over port **10518**, create a `<APP_NAME>.d/conf.yaml` file at the root of your [Agent's configuration directory][1] with the following content:
+To gather logs from your `<APP_NAME>` application that forwards its logs with to TCP port **10518**, create a `<APP_NAME>.d/conf.yaml` file at the root of your [Agent's configuration directory][1] with the following content:
```
Comment (alai97) [inline content/en/agent/logs/_index.md:81]:
```suggestion
To gather logs from your `<APP_NAME>` application that forwards its logs to TCP port **10518**, create a `<APP_NAME>.d/conf.yaml` file at the root of your [Agent's configuration directory][1] with the following content:
```

File: content/en/agent/logs/_index.md [hunk 2]
```diff
@@ -90,6 +90,10 @@ logs:
 
 If you are using Serilog, `Serilog.Sinks.Network` is an option for connecting with UDP.
 
+As of Agent version 7.31.0, the agent will keep a TCP connection open indefinitely.
```
Comment (alai97) [inline content/en/agent/logs/_index.md:93]:
```suggestion
As of Agent version 7.31.0, the Agent keeps a TCP connection open indefinitely.
```

File: content/en/agent/logs/_index.md [hunk 3]
```diff
@@ -90,6 +90,10 @@ logs:
...
+Previous versions closed conections after 60 seconds without log messages.
```
Comment (alai97) [inline content/en/agent/logs/_index.md:94]: This was not documented before; let's not document it now.

```suggestion
```

File: content/en/agent/logs/_index.md [hunk 4]
```diff
@@ -90,6 +90,10 @@ logs:
...
+On connection failure, your application should re-connect to the agent.
```
Comment (alai97) [inline content/en/agent/logs/_index.md:95]:
```suggestion
On connection failure, your application should reconnect to the Agent.
```

**Review by alai97 — CHANGES_REQUESTED** [review]

File: content/en/agent/logs/_index.md (hunk 3 above)
Comment (djmitche) [reply content/en/agent/logs/_index.md:94]: The sentence before is pretty vague without this counterpoint.  Maybe I could add " open indefinitely, even if idle."?

Or, maybe this doesn't need to be documented at all, if what we think of as "removing a feature" is actually "removing a bug" from the user's perspective. WDYT?

**Comment by djmitche** [comment]
Re-requesting review for

> Or, maybe this doesn't need to be documented at all, if what we think of as "removing a feature" is actually "removing a bug" from the user's perspective. WDYT?

File: content/en/agent/logs/_index.md (hunk 3 above)
Comment (alai97) [reply content/en/agent/logs/_index.md:94]: I think if we rephrased the first sentence as "In the Agent version 7.31.0+, the TCP connection stays open indefinitely even when idle." we can delete the rest of this!

File: content/en/agent/logs/_index.md [hunk 5]
```diff
@@ -90,6 +90,10 @@ logs:
 
 If you are using Serilog, `Serilog.Sinks.Network` is an option for connecting with UDP.
 
+As of Agent version 7.31.0, the Agent keeps a TCP connection open indefinitely.
+Previous versions closed conections after 60 seconds without log messages.
+On connection failure, your application should reconnect to the Agent.
```
Comment (alai97) [inline content/en/agent/logs/_index.md:95]:
```suggestion
```

File: content/en/agent/logs/_index.md [hunk 6]
```diff
@@ -90,6 +90,10 @@ logs:
 
 If you are using Serilog, `Serilog.Sinks.Network` is an option for connecting with UDP.
 
+As of Agent version 7.31.0, the Agent keeps a TCP connection open indefinitely.
```
Comment (alai97) [inline content/en/agent/logs/_index.md:93]:
```suggestion
In the Agent version 7.31.0+, the TCP connection stays open indefinitely even when idle.
```

**Review by alai97 — CHANGES_REQUESTED** [review]

**Review by alai97 — APPROVED** [review]
Looks good, thanks for the changes @djmitche!

**Pull request merged**